.github/workflows/test-java.yaml

```diff
@@ -15,11 +15,47 @@ jobs:
   test:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-    runs-on: ${{matrix.os}}
+        # start with the most popular ones
+        include:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+            jdk: 8
+          - os: macos-latest
+            target: aarch64-apple-darwin
+            # https://adoptium.net/supported-platforms/
+            jdk: 11
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+            jdk: 8
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            jdk: 8
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+      - name: Install JDK 8 and Maven
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: ${{ matrix.jdk }}
+          cache: "maven"
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
```

Comment:
Unneeded

```suggestion
```

Reply:
Why? Is matrix.target unneeded too?

Reply:
We need to put the `target` in the `cargo` command, just like you do above: https://github.com/PRQL/prql/pull/1929/files#diff-787cc4dbb3b51e6d39655bfc50bee79b4b6c34a7d629402d34342ebd0e878549R60

Unfortunately `actions-rs/toolchain@v1` isn't maintained, and doesn't bring much benefit above the standard runner now. So we just use the standard runner and pass the target.